docs/design-roadmap/ha-control-plane.md

```diff
@@ -0,0 +1,84 @@
+Whether to [embed](https://pkg.go.dev/github.com/coreos/etcd/embed) an etcd server in k8ssandra-operator and span it across the multiple Kubernetes clusters (which may come in handy in the future but which is a heavy weight dependency). 
+What consensus mechanism to use otherwise (Raft or a Paxos derivative are probably sensible choices).
+
+### State replication
+
+In the current design, only the control-plane cluster has a copy of the K8ssandraCluster resource. But if any k8ssandra-operator instance can be a leader, this means that all instances need to have access to the current cluster target state as represented by the K8ssandraCluster k8s object. 
+
+My first tentative suggestion here is that we will need to replicate the K8ssandraCluster object to all k8ssandra-operator instances (i.e. to all k8s clusters). 
+
+We can distinguish between those that should be acted upon (because they are in the leader's cluster) and those which are present only for resilience using an additional annotation on the resource.
+
+But an additional complication again arises in ensuring consistency. What happens if changes are made to the CR but fail to replicate before control plane failure?
+
+Our options are;
+1. Simply accept that some k8ssandra-operator instances may take over as leader with a slightly stale state and then begin reconciling towards it, in the case where a leader failure has occurred.
+2. Include replication logic in the validating webhook so that all/some minimum number of clusters must have received any updates to the K8ssandraCluster CR before it is accepted by the recipient control plane server. (Think Cassandra's write acknowledgement behavior).
+3. Investigate the use of Kubefed to see if it has functionality that might suit our use case. I would want to be certain on how it fails over in the event of whole-of-cluster loss, as there is little discussion in the Kubefed docs as to what happens in extreme failure scenarios. 
+
+Option 2 sounds attractive but may not be possible under real-world latency constraints - i.e. the HTTP connection between the client (e.g. kubectl) and the API server may timeout before the data can replicate. We would need to examine mitigations for this sort of issue.
+
+## Considerations and outstanding questions
+
+### Additional user stories that may be relevant
+
+This topic touches on a number of operational questions that users are likely to face in the real world. I'm not sure how many of these we have/want to consider, but I thought it'd be good to document them somewhere.
+
+- A user has an existing CassandraDataCenter from Cass operator. How can they join it to a new K8ssandra cluster?
+- What is the process for migrating control plane DCs and data plane DCs to new k8s clusters generally?
+- What cases do we support around running some DCs under k8ssandra and others externally? (e.g. in a traditional bare metal DC.) 
+- Do we support mixed (k8s/bare-metal) clusters only at migration or permanently.
```

Comment:
My opinion there is that we can support them permanently, but the automation we have in k8ssandra-operator won't spill over to the external clusters. Operations spanning over both clusters will have to be handled manually, which can come with some surprises.

Reply:
So I guess we'd recommend against it.

Reply:
I'm happy to recommend against. I suggest we leave the question in the doc though.